0_fund_analysis_functions_v2.R

```diff
@@ -0,0 +1,1210 @@
+  #     .funs = tolower
+  #   )
+  # 
+  # sensitive_sectors <- sensitive_sectors %>% 
+  #   mutate(
+  #     tot_rev = if_else(tot_rev > 100, 100, tot_rev),
+  #     id = as.character(id)
+  #     )
+  # 
+  # sensitive_sectors <- sensitive_sectors %>% 
+  #   bind_rows(revenue_data) %>% 
+  #   distinct_all()
+  
+  # prepare regions 
+  regions <<- read_csv(path(git_path, "data", "regions", ext = "csv"))
+  
+  # indicator selection
+  project_indicator_selection <<- read_csv(path(project_location, "10_Parameter_File", "fund_results_indicator_selection", ext = "csv"))
+  
+  # raw MorningStar data
+  general_fund_data <<- read_csv(path(project_location, "20_Raw_Inputs", paste(project_name, "fund_info", sep = "_"), ext = "csv")) %>% 
+    clean_names(case = "snake")
+  
+  portfolio_results <<- load_pacta_results(project_location) 
+  
+  audit_file <<- load_pacta_portfolio(project_location) 
+  
+  # Ports.Overview <<- read_csv(paste0(project_location, paste(project_name, "overview_portfolio", sep = "_"), ext = "csv"))
+  
+  # TODO: determine company with highest coal % in power sector (use filter of at least 100 MW?)
+  # pacta_company_debt_results <- read_rds(paste0(project_location, "40_Results/Bonds_results_company.rds"))
+  # pacta_company_equity_results <- read_rds(paste0(project_location, "40_Results/Equity_results_company.rds"))
+  # 
+  # pacta_company_debt_results$asset_type <- "Bonds"
+  # pacta_company_equity_results$asset_type <- "Equity"
+  # 
+  # pacta_company_results <- pacta_company_debt_results %>% 
+  #   bind_rows(pacta_company_equity_results)
+  # 
+  # pacta_company_results <- pacta_company_results %>% 
+  #   filter((ald_sector == "Power" & scenario_geography == "GlobalAggregate") | (ald_sector != "Power" & scenario_geography == "Global")
+  #   )
+  # 
+  # pacta_company_results <- pacta_company_results %>% 
+  #   filter(investor_name != portfolio_name | portfolio_name == "Meta Portfolio")
+  # 
+  # pacta_company_results <- pacta_company_results %>% 
+  #   clean_names(case = "snake") %>% 
+  #   mutate_at(
+  #     .vars = c("ald_sector", "technology", "asset_type"),
+  #     .funs = tolower
+  #   )
+  # 
+  # input_pacta_company_results <<- pacta_company_results
+}
+
+
+check_group_share_parameters <- function(
+  .data, 
+  id_cols,
+  numerator_group, 
+  denominator_group,
+  values_from,
+  values_to
+) {
+  character_parameters <- c(numerator_group, denominator_group, values_from)
+  character_null_parameters <- c(id_cols, values_to)
+  # check parameters 
+  lapply(character_parameters, function(x) {stopifnot(is.character(x) & !is.null(x))})
+  lapply(character_null_parameters, function(x) {stopifnot(is.character(x) | is.null(x))})
+  # check names 
+  crucial_names <- c(values_from, numerator_group, denominator_group, id_cols)
+  r2dii.utils::check_crucial_names(.data, crucial_names)
+  # check values
+  assertr::verify(.data, class(.data[[values_from]]) == "numeric")
+  # return 
+  return(.data)
```

Comment:
This function is called primarily for its side effect. Return `invisible(.data)` instead of `return(.data)`.
https://principles.tidyverse.org/out-invisible.html